src/BCrypt/BCrypt+BCRYPT_AUTHENTICATED_CIPHER_MODE_INFO.cs

```diff
@@ -0,0 +1,138 @@
+﻿// Copyright (c) to owners found in https://github.com/AArnott/pinvoke/blob/master/COPYRIGHT.md. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+
+namespace PInvoke
+{
+    using System;
+    using System.Runtime.InteropServices;
+
+    /// <content>
+    /// Contains the <see cref="BCRYPT_AUTHENTICATED_CIPHER_MODE_INFO"/> nested type.
+    /// </content>
+    public partial class BCrypt
+    {
+        /// <summary>
+        /// Used with the <see cref="BCryptEncrypt(SafeKeyHandle, byte[], IntPtr, byte[], BCryptEncryptFlags)"/> and <see cref="BCryptDecrypt(SafeKeyHandle, byte[], IntPtr, byte[], BCryptEncryptFlags)"/> functions
+        /// to contain additional information related to authenticated cipher modes.
+        /// </summary>
+        [StructLayout(LayoutKind.Sequential)]
+        public struct BCRYPT_AUTHENTICATED_CIPHER_MODE_INFO
+        {
+            /// <summary>
+            /// The version of the <see cref="BCRYPT_AUTHENTICATED_CIPHER_MODE_INFO"/> struct.
+            /// </summary>
+            public const uint BCRYPT_AUTHENTICATED_CIPHER_MODE_INFO_VERSION = 1;
+
+            /// <summary>
+            /// The size, in bytes, of this structure.
+            /// Do not set this field directly. Use the <see cref="Create"/> method instead.
+            /// </summary>
+            public uint cbSize;
+
+            /// <summary>
+            /// The version number of the structure.
+            /// The only supported value is <see cref="BCRYPT_AUTHENTICATED_CIPHER_MODE_INFO_VERSION"/>.
+            /// Do not set this field directly. Use the <see cref="Create"/> method instead.
+            /// </summary>
+            public uint dwInfoVersion;
+
+            /// <summary>
+            /// A pointer to a buffer that contains a nonce.
+            /// The Microsoft algorithm providers for Advanced Encryption Standard (AES)
+            /// require a nonce for the Counter with CBC-MAC (CCM) and Galois/Counter Mode (GCM)
+            /// chaining modes, and will return an error if none is present.
+            /// If a nonce is not used, this member must be set to NULL.
```

Comment:
I don't think there is a good way to make it automated but it could be made assisted, also the point is moot if we decide to do pointer only or to codegen the IntPtr variants.